app/routes.py

```diff
@@ -0,0 +1,413 @@
+    if not video_id.isnumeric():
+        return jsonify(None), 400
+    
+    video = Video.query.get(video_id)
+    if video == None:
+        return video_not_found(video_id), 404
+
+    response_body = video.to_dict()
+
+    return jsonify(response_body), 200
+
+
+@videos_bp.route("", methods = ["POST"])
+def post_video():
+    """Creates a new video based on given params"""
+
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return {"details": "Request body must include title."}, 400
+
+    if "release_date" not in request_body:
+        return {"details": "Request body must include release_date."}, 400
+
+    if "total_inventory" not in request_body:
+        return {"details": "Request body must include total_inventory."}, 400
+
+    new_video = Video.from_dict(request_body)
+
+    db.session.add(new_video)
+    db.session.commit()
+
+    response_body = new_video.to_dict()
+
+    return jsonify(response_body), 201
+
+@videos_bp.route("<video_id>", methods = ["PUT"])
+def update_video(video_id):
+    """Returns details about
+    a specific video in the store's inventory
+    """
+
+    video = Video.query.get(video_id)
+    if video is None:
+        return video_not_found(video_id), 404
+
+    form_data = request.get_json()
+    
+    if "title" not in form_data or "total_inventory" not in form_data or "release_date" not in form_data:
+        return jsonify(None), 400
+
+    video.title = form_data["title"]
+    video.total_inventory = form_data["total_inventory"]
+    video.release_date = form_data["release_date"]
+
+    db.session.commit()
+
+    response_body = video.to_dict()
+
+    return jsonify(response_body), 200
+
+
+@videos_bp.route("/<video_id>", methods = ["DELETE"])
+def delete_video(video_id):
+    """Deletes a specific video."""
+
+    video = Video.query.get(video_id)
+    if video is None:
+        return video_not_found(video_id), 404
+    
+    if video.customers:
+        for rental_record in video.rentals:
+            db.session.delete(rental_record)
+        db.session.delete(video)
+        db.session.commit()
+        response_body = {"message" : f"{video.title} and all rental records for that film have been deleted",
+        }
+        return jsonify(response_body), 200
+    
+    response_body = {"id" : video.id}
+
+    db.session.delete(video)
+    db.session.commit()
+
+    return jsonify(response_body), 200
+
+
+# ********************************
+# *** rentals custom endpoints ***
+# ********************************
+@rentals_bp.route("/check-out", methods = ["POST"])
+def post_rentals_check_out():
+    """Checks out a video
+    to a customer and updates the database
+    """
+
+    request_body = request.get_json()
+    if "customer_id" not in request_body:
+        return {"details": "Request body must include customer_id."}, 400
+    
+    if "video_id" not in request_body:
+        return {"details": "Request body must include video_id."}, 400
```

Comment:
We can combine these conditionals

```suggestion
    if "customer_id" not in request_body or  "video_id" not in request_body:
        return {"details": "Request body must include customer_id."}, 400
```